consensus/consensus-types/src/proof_of_store.rs

```diff
@@ -145,6 +146,24 @@ impl Display for BatchInfo {
     }
 }
 
+impl TDataInfo for BatchInfo {
+    fn num_txns(&self) -> u64 {
+        self.num_txns()
+    }
+
+    fn num_bytes(&self) -> u64 {
+        self.num_bytes()
+    }
+
+    fn info(&self) -> &BatchInfo {
+        &self
+    }
+
+    fn signers(&self, _ordered_authors: &Vec<PeerId>) -> Vec<PeerId> {
+        vec![self.author()]
```

Comment:
are we adding the block proposer to this somewhere as well?

Reply:
yes. i've added it in payload manager.